android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/Addon.kt

```diff
@@ -280,35 +291,73 @@ data class Addon(
         fun newFromWebExtension(extension: WebExtension, installedState: InstalledState? = null): Addon {
-            val name = extension.getMetadata()?.name ?: extension.id
-            val description = extension.getMetadata()?.description ?: extension.id
-            val permissions = extension.getMetadata()?.permissions.orEmpty() +
-                extension.getMetadata()?.hostPermissions.orEmpty()
-
-            val developerName = extension.getMetadata()?.developerName.orEmpty()
+            val metadata = extension.getMetadata()
+            val name = metadata?.name ?: extension.id
+            val description = metadata?.description ?: extension.id
+            val permissions = metadata?.permissions.orEmpty() +
+                metadata?.hostPermissions.orEmpty()
+
+            val averageRating = metadata?.averageRating ?: 0f
+            val reviewCount = metadata?.reviewCount ?: 0
+            val siteUrl = metadata?.homePageUrl.orEmpty()
+            val reviewUrl = metadata?.reviewUrl.orEmpty()
+            val developerName = metadata?.developerName.orEmpty()
             val author = if (developerName.isNotBlank()) {
-                Author(name = developerName, url = extension.getMetadata()?.developerUrl.orEmpty())
+                Author(name = developerName, url = metadata?.developerUrl.orEmpty())
             } else {
                 null
             }
 
             return Addon(
                 id = extension.id,
                 author = author,
-                version = extension.getMetadata()?.version.orEmpty(),
+                version = metadata?.version.orEmpty(),
                 permissions = permissions,
-                downloadUrl = extension.url,
-                siteUrl = extension.url,
-                translatableName = mapOf(Addon.DEFAULT_LOCALE to name),
-                translatableDescription = mapOf(Addon.DEFAULT_LOCALE to description),
+                downloadUrl = metadata?.downloadUrl.orEmpty(),
+                rating = Rating(averageRating, reviewCount),
+                siteUrl = siteUrl,
+                translatableName = mapOf(DEFAULT_LOCALE to name),
+                translatableDescription = mapOf(DEFAULT_LOCALE to metadata?.fullDescription.orEmpty()),
                 // We don't have a summary when we create an add-on from a WebExtension instance so let's
                 // re-use description...
-                translatableSummary = mapOf(Addon.DEFAULT_LOCALE to description),
-                updatedAt = "",
+                translatableSummary = mapOf(DEFAULT_LOCALE to description),
+                updatedAt = fromMetadataToAddonDate(metadata?.updateDate.orEmpty()),
+                reviewUrl = reviewUrl,
                 installedState = installedState,
             )
         }
 
+        /**
+         * Returns a new [String] formatted in "yyyy-MM-dd'T'HH:mm:ss'Z'".
+         * [Metadata] uses "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'" while [Addon] uses "yyyy-MM-dd'T'HH:mm:ss'Z'"
```

Comment:
We could mention that the metadata date format is the "simplified 8601 format" here maybe.